Newspost: New Changes to Star Rating and PP (#2000)
ppy/osu-wiki: news/2019-01-30-new-changes-to-star-rating-performance-points.md

---
layout: post
title: "New Changes to Star Rating and Performance Points"
date: 2019-01-30 01:00:00 +0000
---

After many years of stagnancy, some changes are finally coming to the way Star Rating and Performance Points are calculated! Read on to find out more.

![](temporary.banner)

Hello!

> It is with great anticipation that I, Xexxar, am here to announce changes to both Peformance Points (aka: pp) and the star rating system. 

Reviewer: ```suggestion
It is with great anticipation that I, Xexxar, am here to announce changes to both Performance Points (aka: pp) and the star rating system. 
```

> It is with great anticipation that I, Xexxar, am here to announce changes to both Performance Points (aka: pp) and the star rating system. 

Reviewer 2: This reads rather weirdly, since it leaves a lot to imagination exactly what you're anticipating, considering you're in the process of doing the announcement already.

Maybe simply say something about this being a long time coming, or "with great excitement".

Author: Changed.

I have been closely involved with various other members of the community over the past few months in proposing amendments to the Performance Points system with the end goal of updating a core part of osu!'s competitive progression that hasn't seen any significant changes in many, many years. The game has evolved considerably since the last update to these key systems, with the introduction of many new and different styles of mapping and gameplay.

While the updates we've just made are not designed to fix *every* issue with the systems as they currently stand, we hope that the changes will allow players to better assess skills that were formerly not factored in properly in previous versions of the algorithm.

Without further delay, let's get into the specific changes:

## Wide Angle Jumps

> As many players are well aware, the angle of a jump impacts how difficulty a pattern is to combo. Evidence of this can be found throughout modern mapping. Almost all maps that are popular for their PP values contain jumps that are typically non-linear.

Reviewer 3: s/difficulty/difficult

Author: fixed

> As many players are well aware, the angle of a jump impacts how difficulty a pattern is to combo. Evidence of this can be found throughout modern mapping. Almost all maps that are popular for their PP values contain jumps that are typically non-linear.

Reviewer 2: ```suggestion
As many players are well aware, the angle of a jump impacts how difficulty a pattern is to combo. Evidence of this can be found throughout modern mapping, where almost all maps that are popular for their PP values contain jumps that are typically non-linear.
```

Author: Changed.

> As many players are well aware, the angle of a jump impacts how difficulty a pattern is to combo. Evidence of this can be found all throughout modern mapping. Almost all maps that are popular for their PP values contain jumps that are typically non-linear. 

Reviewer 4: ```suggestion
As many players are well aware, the angle of a jump impacts how difficulty a pattern is to combo. Evidence of this can be found throughout modern mapping. Almost all maps that are popular for their PP values contain jumps that are typically non-linear.
```

As such, through the usage of a new angle calculation, a new scaling reward for jumps with angles exceeding 60 degrees has been added. 

> Decided not to farm a map because it had square jumps? Found yourself baffled by the difficulty of an older 2013 map with linear flow? These previously underweighted maps now have a more accurate star rating associated with them (this includes alt maps). And yes, they will be worth more PP as a result.

Reviewer 2: Imo, assume your reader doesn't know what an alt map is and give a brief explanation.

Author: Provided an example as well.

> Decided not to farm a map because it had square jumps? Found yourself baffled by the difficulty of an older 2013 map with linear flow? These previously underweighted maps now have a more accurate star rating associated with them (this includes alt maps). And yes, they will be worth more PP as a result.

Reviewer 4: ```suggestion
Decided not to farm a map because it had square jumps? Found yourself baffled by the difficulty of an older 2013 map with linear flow? These previously underweighted maps now have a more accurate star rating associated with them (this includes alt maps). Also, yes, they will be worth more PP as a result.
```

Author: I think this is fine the way it is.

## Streams and High BPM

> Besides the changes to aim, there has been an adjustment to how streams are handled by the algorithm. Many players who are capable of incredible speeds find their ranks are unfairly low given their unique skillset. In order to promote these unique skillsets, streams exceeding 200 BPM now have an increased reward that more accurately represents their difficulty.

Reviewer 4: ```suggestion
Besides the changes to aim, there has been an adjustment to how streams are handled by the algorithm. Many players who are capable of incredible speeds find their ranks are unfairly low given their unique skillset. To promote these unique skillsets, streams exceeding 200 BPM now have an increased reward that more accurately represents their difficulty.
```

Able to FC 300 BPM stacks with your laptop keyboard? This is the change you've been waiting for. 

> In addition, a small nerf to very spaced streams has been introduced. These are those streams found in maps such as Honesty and Sidetracked Day. While the nerf is sizable, these plays are still quite impressive and are still worth a lot, just not quite as much as they were worth before.

Reviewer 2: This is nitpicky, but you are using the joining statement "in addition" despite leading in with the previous sentence.

Reviewer 2: Link to the respective maps using markdown

Reviewer 2: "Sizable" and "small nerf" are contradicting each other, maybe replace sizable with "noticeable"

> In addition, a small nerf to very spaced streams has been introduced. These are those streams found in maps such as Honesty and Sidetracked Days. While the nerf is sizable, these plays are still quite impressive and are still worth a lot, just not quite as much as they were worth before.

Reviewer 4: ```suggestion
In addition, a small nerf to very spaced streams has been introduced. These are those streams found in maps such as Honesty and Sidetracked Day. While the nerf is sizable, these plays are still quite impressive and are still worth a lot, just not quite as much as they were worth before.
```

## Slider Buffs

> Are you a fan of playing those impressive and previously underweighed tech maps with high slider velocity? 

Reviewer 2: I'd say the same to assume players don't know what tech maps are, but the definition is up for debate I guess.

Well, say no more! 

> While the specifics are quite complicated, in general, these fast sliders are now more accurately judged by the algorithm. Not only that, but maps with long sliders that have jumps between them also receive a large buff compared to before.

Reviewer 2: ```suggestion
While the specifics are quite complicated, in general, these fast sliders are now more accurately judged by the algorithm. Not only that, but maps with long sliders that have jumps between them also received a large buff compared to before.
```

> While the specifics are quite complicated, in general these fast sliders are now more accurately judged by the algorithm. Not only that, but maps with long sliders that have jumps between them also receive a large buff compared to before. 

Reviewer 4: ```suggestion
While the specifics are quite complicated, in general, these fast sliders are now more accurately judged by the algorithm. Not only that, but maps with long sliders that have jumps between them also receive a large buff compared to before.
```

> This is a change that will help buff many tech maps with fast gimmicky sliders. While this change will not be noticeable on many maps, the ones that do benefit may see substantial buffs!

Reviewer 2: Imo, help differentiate which maps are not affected, since you state that it will buff many tech maps and at the same time will not be noticeable on many maps.

Again, will need a definition for tech maps to make this happen.

Author: I believe my edit should help this make more sense.

## Flashlight Rework

> One of the exciting introductions that will be coming with this patch is the change to flashlight. Primarily a dead mod outside of farming 30 second maps, flashlight has been irrelevant since PPV2's inception back in 2014. 

Reviewer 2: consider linking to relevant resources regarding ppv2

Author: Done.

> One of the exciting introductions that will be coming with this patch is the change to flashlight. Primarily a dead mod outside of farming 30 second maps, flashlight has been irrelevant since PPV2's inception back in 2014. 

Reviewer 4: One of the exciting introductions that will be coming with this patch is the change to flashlight. Primarily a dead mod outside of farming 30-second maps, flashlight has been irrelevant since PPV2's inception back in 2014.
```suggestion
One of the exciting introductions that will be coming with this patch is the change to flashlight. Primarily a dead mod outside of farming 30-second maps, flashlight has been irrelevant since PPV2's inception back in 2014.
```

Author: I don't really think this is required.

> In order to remedy this, flashlight will now scale at a much higher rate with the length of a map. To compensate for this increased buff for longer maps, the shorter maps that were previously played with flashlight will receive a small nerf.

Reviewer 4: ```suggestion
To remedy this, flashlight will now scale at a much higher rate with the length of a map. To compensate for this increased buff for longer maps, the shorter maps that were previously played with flashlight will receive a small nerf.
```

## Hidden Rework

> Yes, HD is changing again. With the introduction of the high BPM buff, we wanted to make changes to HD so that maps with fast stacked streams weren't overly buffed when HD is applied. 

Reviewer 2: consider linking to previous changes for people who have no context

Author: there is no previous post, so I did not do this.

> Yes, HD is changing again. With the introduction of the high BPM buff, we wanted to make changes to HD so that maps with fast stacked streams weren't overly buffed when HD is applied. 

Reviewer 2: nitpicky but
```suggestion
Yes, HD is changing once again. With the introduction of the high BPM buff, we wanted to make changes to HD so that maps with fast stacked streams weren't overly buffed when HD is applied. 
```

Author: done.

> Instead of the old system, HD now rewards accuracy more, and both aim and speed PP will scale off of the approach rate of the map. 

Reviewer 2: Consider giving context to what the old system was like. Also, this sentence reads awkwardly, consider something like
```suggestion
Instead of the old system, HD now rewards accuracy more, with both aim and speed bonuses scaling from the approach rate of the map. 
```

Author: Done.

> Find yourself good at farming EZHD plays? You'll find a notable increase in performance points for these plays after the update. 

Reviewer 3: the term "farming" probably doesn't apply here very well, i'd just change to `Are you good at playing with EZ and HD?`

Author: fixed

## Misc PP Changes

These are the more minor technical changes that we introduced to keep things balanced. 

> To combat potentially unimpressive keyboard mashing that may be common for high speed plays as a result of the new buff, a new reward curve for accuracy has been introduced for the speed skill.

Reviewer 2: ```suggestion
To combat potentially unimpressive keyboard mashing that may be common for high speed plays as a result of the new buffs, a new reward curve for accuracy has been introduced for the speed skill.
```

Reviewer 2: Also differentiating between buffs and nerfs seems wrong. IMO replace every instance with "changes".

> In order to combat potentially unimpressive keyboard mashing that may be common for high speed plays as a result of the new buff, a new reward curve for accuracy has been introduced for the speed skill. 

Reviewer 4: ```suggestion
To combat potentially unimpressive keyboard mashing that may be common for high speed plays as a result of the new buff, a new reward curve for accuracy has been introduced for the speed skill.
```
